R/drive_list.R

```diff
@@ -124,12 +128,32 @@ get_leafmost_id <- function(path) {
   )
   child_is_present <- purrr::map_lgl(
     folders$id,
-    ~ .x %in% folders$parents | .x == folder
+    ~ .x %in% folders$parents | .x %in% folder
   )
+
+  ## par down, now we know all but the final layer
   folders <- folders[parent_is_present & child_is_present, ]
+
+  ## I have to rerun this because if there are x folders named foo and
+  ## the one we are interested in is in the root, we will have multiple
+  ## in "folder", but just want one.
+  folder <- folders$id[folders$depth == d]
+
+  # if there are multiple in depth d & it isn't the root
+  if (length(folder) > 1) {
+    leafmost_parent <- folders[folders$depth == d - 1, ]
+    child_is_leafmost <- purrr::map_lgl(
+      folders$parents,
+      ~ .x == leafmost_parent$id
+    )
+    folder <- folders$id[child_is_leafmost]
+  }
```

Comment:
I'm a bit foggy on the scenario we are guarding against here, but I don't disagree. Would be good to write it down specifically and incorporate into a test. All of this will be much easier to code and discuss when there are tests with concrete scenarios for directory structure and desired result.